src/main/java/nextstep/subway/section/domain/Sections.java

```diff
@@ -9,27 +12,71 @@
 import java.util.List;
 import java.util.stream.Collectors;
 
+import static nextstep.subway.exception.ApiExceptionMessge.EXISTS_ALL_STATIONS;
+import static nextstep.subway.exception.ApiExceptionMessge.NOT_EXISTS_STATIONS;
+
 @Embeddable
 public class Sections {
 
 	@OneToMany(mappedBy = "line", cascade = CascadeType.ALL, orphanRemoval = true)
 	private List<Section> sections = new ArrayList<>();
 
-
 	public void add(final Section section) {
-		sections.add(section);
+		if (CollectionUtils.isEmpty(sections) == false) {
+			validateStations(section);
+			connectIfExistsUpStation(section);
+			connectIfExistsDownStation(section);
+		}
+		this.sections.add(section);
 	}
 
 	public boolean contain(final Section section) {
 		return sections.contains(section);
 	}
 
+	private void validateStations(final Section section) {
+		boolean isExistsUpStation = containUpStation(section);
+		boolean isExistsDownStation = containDownStation(section);
+		if (isExistsUpStation && isExistsDownStation) {
+			throw new ApiException(EXISTS_ALL_STATIONS);
+		}
+		if (isExistsUpStation == false && isExistsDownStation == false) {
```

Comment:
```suggestion
		if (!isExistsUpStation && !isExistsDownStation) {
```
더 짧게 쓸 수 있는데 false로 비교하도록 만들 필요가 있을까요? 
!를 놓치는 경우가 있어서 false를 직접 비교하도록 만들었다면 테스트 코드에 해당 케이스를 추가해서 변경이 일어났을 때 테스트 실패하도록 만들면 되지 않을까요?

Reply:
! << 를 사용하는 구분이 쉽게 보이지 않아서 == false 비교를 이용하는 컨벤션으로 현업에서 사용하다보니 나오는 습관인 것 같습니다.
현구님과 같이 보는 프로젝트라고 생각해 부정비교는 앞으로 '!'로 진행해보겠습니다! ㅎㅎ